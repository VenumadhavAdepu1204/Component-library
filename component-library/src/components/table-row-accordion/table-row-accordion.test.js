import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import TableRowAccordion from './index'

describe('TableRowAccordion', () => {
  afterEach(cleanup)
  test('renders properly', () => {
    const titleContent = <p>Title content</p>
    const bodyContent = <p>Body content</p>
    const { container } = render(<TableRowAccordion titleContent={titleContent} bodyContent={bodyContent} />)
    expect(container).toMatchSnapshot()
  })

  test('link calls handleClick on click', () => {
    const titleContent = <p>Title content</p>
    const bodyContent = <p>Body content</p>
    const { getByTestId } = render(<TableRowAccordion titleContent={titleContent} bodyContent={bodyContent} />)
    const accordionButton = getByTestId('table-row-accordion')
    fireEvent.click(accordionButton)
  })
})
