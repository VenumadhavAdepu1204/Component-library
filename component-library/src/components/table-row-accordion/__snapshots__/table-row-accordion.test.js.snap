// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRowAccordion renders properly 1`] = `
<div>
  <div
    class="sc-bdVaJa hUFsxd"
  >
    <button
      class="sc-bxivhb blLPzl"
      data-testid="table-row-accordion"
    >
      <svg
        data-test-id="undefined-icon-svg"
        focusable="false"
        height="1em"
        viewBox="0 0 32 32"
        width="1em"
      >
        <path
          d="M25.4,10.6 C24.5,9.8 23.2,9.8 22.3,10.6 L16,16.9 L9.7,10.6 C8.8,9.8 7.5,9.8 6.6,10.6 C5.7,11.4 5.7,12.8 6.6,13.7 L14.4,21.5 C15.3,22.3 16.6,22.3 17.5,21.5 L25.3,13.7 C26.2,12.8 26.2,11.5 25.4,10.6 Z"
          data-test-id="undefined-icon-path"
          fill="#0093D8"
          id="undefined-data-id"
        />
      </svg>
    </button>
    <div
      class="sc-bwzfXH gVzjuy"
    />
  </div>
</div>
`;
